### PR TITLE
Mint Components update for RingCentral

### PR DESCRIPTION
**Pull request by noahwc**
## Description of the change

- Add "simple" variant of the referral table rewards column (no expanding)
- Add loading states to sqm-share-button
- Improve sqm-big-stat paths to avoid `null` in paths
- Tests added for a variety of sqm-big-stat statTypes

## Type of change
- [x] Bug fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Documentation or Development tools (readme, specs, tests, code formatting)

## Links

 - Jira issue number: https://saasquatch.atlassian.net/browse/DEV-3094
 - Process.st launch checklist: https://app.process.st/runs/RingCentral-employee-portal-qKlCZi1BYEORSuq7_9FD_A

### Development

- [ ] [Prettier](https://www.npmjs.com/package/prettier) was run
- [ ] The behaviour changes in the pull request are covered by specs
- [ ] All tests related to the changed code pass in development

### Paperwork

- [ ] This pull request has a descriptive title and information useful to a reviewer
- [ ] This pull request has a Jira number
- [ ] This pull request has a Process.st launch checklist
- [ ] "Ready for review" label attached to the PR and reviewers pinged on Slack

### Code review 

- [ ] Changes have been reviewed by at least one other engineer
- [ ] Security impacts of this change have been considered

### CODE REVIEW

**Comment by DerekSiemens** [comment]
Seems like changelog is missing some of the updates that Zach made, but those might actually not be in this PR - We should probably centralize all those changes and update the change log here

**Comment by zachharrison** [comment]
> Seems like changelog is missing some of the updates that Zach made, but those might actually not be in this PR - We should probably centralize all those changes and update the change log here

Added those changes

File: packages/mint-components/src/components/sqm-leaderboard/readme.md [hunk 1]
```diff
@@ -7,14 +7,15 @@
...
+| `statsheading`    | `statsheading`     |                                                            | `string`                                                                                                                                                                                                                                                                                      | `undefined` |
+| `unit`            | `unit`             | Unit used with leaderboard type "singleUnitRewardValueSum" | `string`                                                                                                                                                                                                                                                                                      | `undefined` |
```
Comment (loganvolkers) [inline packages/mint-components/src/components/sqm-leaderboard/readme.md:17]: Rebuild readme

File: packages/mint-components/CHANGELOG.md [hunk 2]
```diff
@@ -7,6 +7,28 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
...
+
+- Filtered Credit and Integration reward count stats now available in \<sqm-big-stat>
+  - formats:
+  ```
+  /(rewardsCountFiltered)/:statType([INTEGRATION|PCT_DISCOUNT|CREDIT]*)?/:unit((?!global)(?!PENDING)(?!CANCELLED)(?!EXPIRED)(?!REDEEMED)(?!AVAILABLE)[a-zA-Z0-9%]+)?/:status([PENDING|CANCELLED|EXPIRED|REDEEMED|AVAILABLE]*)?/:global?
+  /(integrationRewardsCountFiltered)/:status([PENDING|CANCELLED|EXPIRED|REDEEMED|AVAILABLE]*)?/:global?
+  ```
+- \<sqm-share-button> loading states and hide text prop
+- \<sqm-divided-layout> border customizable
+- \<sqm-leaderboard> hook stories created
```
Comment (loganvolkers) [inline packages/mint-components/CHANGELOG.md:29]:
```suggestion
- \<sqm-leaderboard> allows rank to be shown
```

File: packages/mint-components/CHANGELOG.md [hunk 3]
```diff
@@ -7,6 +7,28 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
...
+- \<sqm-big-stat> e2e tests created
```
Comment (loganvolkers) [inline packages/mint-components/CHANGELOG.md:30]:
```suggestion
```

File: packages/mint-components/CHANGELOG.md [hunk 4]
```diff
@@ -7,6 +7,28 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
...
 
+## [1.2.1] - 2021-09-21
+
+### Fixed
+
+- demoData no longer breaking GrapesJS editability on all components
+- \<sqm-big-stat> properly supports units with/without slashes
+- \<sqm-big-stat> rewardBalance stat now uses format if included for the fallback value
+- \<sqm-referral-table> bugs fixed
+
```
Comment (loganvolkers) [inline packages/mint-components/CHANGELOG.md:18]: Note on `menuLabel` removed from sqm-program-menu

File: packages/mint-components/src/components/sqm-referral-table/cells/sqm-referral-table-rewards-cell.tsx [hunk 5]
```diff
@@ -106,21 +116,29 @@ export class ReferralTableRewardsCell {
...
+                <sl-badge
+                  class={sheet.classes.StatusBadge}
+                  type={slBadgeType}
+                  pill
+                >{`${badgeText} ${
                   getTimeDiff(reward.dateScheduledFor) === "tomorrow" ||
                   getTimeDiff(reward.dateScheduledFor) === "today" ||
-                  getTimeDiff(reward.dateExpires) === "next month" ||
-                  getTimeDiff(reward.dateExpires) === "next year"
+                  getTimeDiff(reward.dateScheduledFor) === "next month" ||
```
Comment (loganvolkers) [inline packages/mint-components/src/components/sqm-referral-table/cells/sqm-referral-table-rewards-cell.tsx:126]: This should just be `Intl` for translation

File: packages/mint-components/src/components/sqm-referral-table/columns/sqm-referral-table-date-column.tsx [hunk 6]
```diff
@@ -3,12 +3,29 @@ import { Component, h, Host, Method, Prop } from "@stencil/core";
...
+   * @uiEnum ["dateConverted",
+   * "dateReferralStarted",
+   * "dateFraudChecksCompleted",
+   * "dateModerated",
+   * "dateModified",
+   * "dateReferralEnded",
+   * "dateReferralPaid",
+   * "dateUserModified"]
+   */
   @Prop() dateShown: string = "dateConverted";
```
Comment (loganvolkers) [inline packages/mint-components/src/components/sqm-referral-table/columns/sqm-referral-table-date-column.tsx:29]: Could update the typescript type to be explicit `"dateConverted" |  "dateReferralStarted" | ...`

File: packages/mint-components/src/components/sqm-referral-table/readme.md [hunk 7]
```diff
@@ -1,4 +1,4 @@
-# sqm-leaderboard
+# sqm-table-row
```
Comment (loganvolkers) [inline packages/mint-components/src/components/sqm-referral-table/readme.md:1]:
```suggestion
# sqm-referral-table
```

File: packages/mint-components/src/components/sqm-referral-table/cells/sqm-referral-table-rewards-cell.tsx (hunk 5 above)
Comment (00salmon) [reply packages/mint-components/src/components/sqm-referral-table/cells/sqm-referral-table-rewards-cell.tsx:126]: We can probably make this a more re-usable utility function, but intl's version is a bit underwhelming, as you still need to calculate which of hours/days/month etc it should use (see https://stackoverflow.com/questions/67373795/how-to-use-intl-formatrelativetime-in-the-correct-way)

**Review by 00salmon — APPROVED** [review]
ready for liftoff

**Pull request merged**